anonlink/blocking.py

```diff
@@ -0,0 +1,238 @@
+from itertools import chain, product, repeat
+from numbers import Real
+from random import Random
+from typing import (Any, Callable, Hashable, Iterable, List,
+                    Optional, overload, Sequence, Tuple, TypeVar, Union)
+
+
+BlockingFunction = Callable[[int, int, Sequence[bool]], Iterable[Hashable]]
+
+
+_T = TypeVar('_T')
+def _evalf(__funcs : Iterable[Callable[..., _T]],
+           *args,
+           **kwargs) -> Iterable[_T]:
+    return (f(*args, **kwargs) for f in __funcs)
+
+
+class block_and:
+    """ Conjunction of multiple blocking functions.
+
+        Records share a block if they share a block in all of the
+        functions.
+
+        :param funcs: Functions whose conjunction we return.
+
+        :return: The blocking function.
+    """
+    __slots__ = '_funcs'
+
+    @overload
+    def __init__(self,
+                 funcs: Iterable[BlockingFunction]
+                 ) -> None:
+        pass
+
+    @overload
```

Comment:
Well that is cool, I did not know this thing. Just to help future contributors can you add a small comment with link.

Reply:
Done! Link to http://mypy.readthedocs.io/en/latest/function_overloading.html